src/services/firebaseService/FirebaseService.ts

```diff
@@ -0,0 +1,99 @@
+import {
+    getDatabase,
+    ref,
+    push,
+    set,
+    update,
+    remove,
+    get,
+    DataSnapshot,
+} from 'firebase/database';
+import { initializeApp } from 'firebase/app';
+import { getAuth } from 'firebase/auth';
+import { TaskType, UpdateTaskParams } from '../../components/ToDoList/types';
+
+const firebaseConfig = {
+    apiKey: process.env.FIREBASE_API_KEY,
+    authDomain: process.env.FIREBASE_AUTH_DOMAIN,
+    projectId: process.env.FIREBASE_PROJECT_ID,
+    storageBucket: process.env.FIREBASE_STORAGE_BUCKET,
+    messagingSenderId: process.env.FIREBASE_MSG_SENDER_ID,
+    appId: process.env.FIREBASE_APP_ID,
+    databaseURL: process.env.FIREBASE_DB_URL,
+};
+
+// Initialize Firebase
+const app = initializeApp(firebaseConfig);
+const database = getDatabase(app);
+export const auth = getAuth(app);
+
+const firebaseService = {
+    getAllTodos: async (): Promise<TaskType[]> => {
+        const todosRef = ref(database, 'todos');
+        const snapshot = await get(todosRef);
+        const todos: TaskType[] = [];
+
+        if (snapshot.exists()) {
+            snapshot.forEach((childSnapshot: DataSnapshot) => {
+                todos.push({ id: childSnapshot.key, ...childSnapshot.val() });
+            });
+        }
+
+        return todos;
+    },
+    createTodo: async (todo: TaskType) => {
+        const todosRef = ref(database, `todos/${todo.id}`);
+        await set(todosRef, todo);
+        return todo;
+    },
+    updateTodo: async (params: UpdateTaskParams): Promise<TaskType | null> => {
+        const { id, description, isCompleted, isEditing } = params;
+        const todoRef = ref(database, `todos/${id}`);
+
+        // Creating an object with the fields to update
+        const updates: Partial<UpdateTaskParams> = {};
+
+        if (description !== undefined) {
```

Comment:
never use this condition.

1. If necessary, always try to set the default value 
2. Never use undefined in code
3. Try to avoid the "if" block as much as possible, or reduce them to a minimum amount